py/server/deephaven/table.py

```diff
@@ -55,13 +60,282 @@
+
+    def with_node_operations(self, recorders: List[RollupNodeOperationsRecorder]) -> RollupTable:
+        """Returns a new RollupTable that will apply the recorded node operations to nodes when gathering
+        snapshots.
+
+        Args:
+            recorders (List[RollupNodeOperationsRecorder]): a list of RollupNodeOperationsRecorder containing
+                the node operations to be applied, they must be ones created by calling the 'node_operation_recorder'
+                method on the same table.
+
+        Returns:
+            a new RollupTable
+
+        Raises:
+            DHError
+        """
+        try:
+            return RollupTable(
+                j_rollup_table=self.j_rollup_table.withNodeOperations(
+                    [op.j_node_ops_recorder for op in recorders]),
+                include_constituents=self.include_constituents, aggs=self.aggs, by=self.by)
+        except Exception as e:
+            raise DHError(e, "with_node_operations on RollupTable failed.") from e
+
+    def with_filters(self, filters: Union[str, Filter, Sequence[str], Sequence[Filter]]) -> RollupTable:
+        """Applies a set of filters to the group-by columns of this RollupTable to produce and return a new
+        RollupTable
+
+        Args:
+            filters (Union[str, Filter, Sequence[str], Sequence[Filter]], optional): the filter condition
+                expression(s) or Filter object(s)
+
+        Returns:
+            a new RollupTable
+
+        Raises:
+            DHError
+        """
+        try:
+            filters = to_sequence(filters)
+            if filters and isinstance(filters[0], str):
+                filters = Filter.from_(filters)
+            filters = j_array_list(filters)
+
+            return RollupTable(j_rollup_table=self.j_rollup_table.withFilters(filters),
+                               include_constituents=self.include_constituents, aggs=self.aggs, by=self.by)
+        except Exception as e:
+            raise DHError(e, "with_filters operation on RollupTable failed.") from e
+
+
+class TreeNodeOperationsRecorder(JObjectWrapper, _FormatOperationsRecorder,
+                                 _SortOperationsRecorder, _FilterOperationsRecorder):
+    """Recorder for node-level operations to be applied when gathering snapshots of TreeTable. Supported operations
+    include column formatting, sorting, and filtering.
+
+    Note: It should not be instantiated directly. User code must call :meth:`~TreeTable.node_operation_recorder` to
+    create an instance of the recorder.
+    """
+
+    j_object_type = _JTreeTableNodeOperationsRecorder
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_node_ops_recorder
+
+    def __init__(self, j_node_ops_recorder: jpy.JType):
+        self.j_node_ops_recorder = j_node_ops_recorder
+
+
+class TreeTable(JObjectWrapper):
+    """ A TreeTable is generated as a result of applying the :meth:`~Table.tree` method on a
```

Comment:
This does not provide a conceptual description of the class/object.  It only describes how it is created.